### PR TITLE
Added logic in Dockerfile and gen_dockerfile.py to support aarch64 security compilation fla…

### PR DESCRIPTION
**Pull request by looptc**
…gs equivalent to the actual ones.

Hi!

As we were talking in the issure, here is the PR.

I sent another PR to cmake_helpers because some .cmake files need also a modification.

Thanks!

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -42,7 +42,7 @@
 	url = https://github.com/desktop-app/lib_storage.git
 [submodule "cmake"]
 	path = cmake
-	url = https://github.com/desktop-app/cmake_helpers.git
+	url = git@github.com:looptc/cmake_helpers.git
```
Comment (ilya-fedin) [inline .gitmodules:45]: This doesn't look right

File: Telegram/build/docker/centos_env/Dockerfile [hunk 2]
```diff
@@ -2,7 +2,12 @@
 {%- set GIT_FREEDESKTOP = GIT ~ "/gitlab-freedesktop-mirrors" -%}
 {%- set QT = "6.7.2" -%}
 {%- set QT_TAG = "v" ~ QT -%}
+{%- set CFLAGS = "-O3 -pipe -fPIC -fno-omit-frame-pointer -fstack-protector-all -fstack-clash-protection -fcf-protection -DNDEBUG -D_FORTIFY_SOURCE=3 -D_GLIBCXX_ASSERTIONS" -%}
 {%- set CFLAGS_DEBUG = "-g -pipe -fPIC -fstack-protector-all -fstack-clash-protection -fcf-protection -D_GLIBCXX_ASSERTIONS" -%}
+{%- if ARCH == "aarch64" -%}
+    {%- set CFLAGS = CFLAGS | replace("-fcf-protection", "-msign-return-address=all -mbranch-protection=standard") -%}
```
Comment (ilya-fedin) [inline Telegram/build/docker/centos_env/Dockerfile:8]: 1. Isn't -msign-return-address deprecated by -mbranch-protection?
2. Isn't `-mbranch-protection=standard` the default?

File: Telegram/build/docker/centos_env/gen_dockerfile.py [hunk 3]
```diff
@@ -3,11 +3,14 @@
...
 from jinja2 import Environment, FileSystemLoader
 
+import platform
+
 def checkEnv(envName, defaultValue):
     return bool(len(environ[envName])) if envName in environ else defaultValue
 
 def main():
     print(Environment(loader=FileSystemLoader(dirname(__file__))).get_template("Dockerfile").render(
+	ARCH=platform.machine(),
```
Comment (ilya-fedin) [inline Telegram/build/docker/centos_env/gen_dockerfile.py:13]: Can't we do it in a similar fashion to target_compile_options_if_exists? Try `echo 'int main() {return 0;}' | g++ -o /dev/null -x c++ -fcf-protection -`, if it fails for you then you can use it to check whether the argument is supported.

File: Telegram/build/docker/centos_env/Dockerfile (hunk 2 above)
Comment (looptc) [reply Telegram/build/docker/centos_env/Dockerfile:8]: Okay, so I remove the flags because they're by default on.

File: .gitmodules (hunk 1 above)
Comment (looptc) [reply .gitmodules:45]: Sorry, I commit the mistake to share the pull request from the same repository where I am using my personal fork. Let me fix that.

File: Telegram/build/docker/centos_env/gen_dockerfile.py (hunk 3 above)
Comment (looptc) [reply Telegram/build/docker/centos_env/gen_dockerfile.py:13]: I have been pulling submodules from master and dev branches, and webrtc is requesting openh264, which needs make OS=linux ARCH=aarch64 to compile. I can remove this, leaving only a removement of the flag for the compiler following your fashion, but maybe in a future we need to have the keyword 'aarch64' in some place accesible for jinja2. What do you think?

Comment (ilya-fedin) [reply Telegram/build/docker/centos_env/gen_dockerfile.py:13]: Let's don't solve problems we haven't encountered yet

Comment (looptc) [reply Telegram/build/docker/centos_env/gen_dockerfile.py:13]: Okay, give me some time to fix this PR.

Comment (looptc) [reply Telegram/build/docker/centos_env/gen_dockerfile.py:13]: I've created and exported a function to later user with jinja2, it runs the command you suggested with the option one by one and sets a final variable. Is this way the right one?

File: Telegram/build/docker/centos_env/Dockerfile (hunk 2 above)
Comment (looptc) [reply Telegram/build/docker/centos_env/Dockerfile:8]: There's a problem if we use -mbranch-protection=pac-ret+leaf with dav1d, it doesn't support leaf functions. May we use an specific compiling options for dav1d, for example -mbranch-protection=standard that works great, or may we use 'standard' for the whole project?

File: Telegram/build/docker/centos_env/gen_dockerfile.py (hunk 3 above)
Comment (ilya-fedin) [reply Telegram/build/docker/centos_env/gen_dockerfile.py:13]: Looks like you haven't pushed it? I don't see any changes

Comment (looptc) [reply Telegram/build/docker/centos_env/gen_dockerfile.py:13]: Sorry, I sent to review but I just wanted to send a simple comment. I'm preparing the docker image, it's finishing soon, then I send changes when it works.

**Pull request closed without merging**